app/factories/AuditEventFactory.scala

```diff
@@ -54,6 +54,26 @@ case class AuditEventFactory() {
     )
   }
 
+  def createMaintainProfileEvent(
+    traderProfile: TraderProfile,
+    updatedTraderProfile: TraderProfile,
+    affinityGroup: AffinityGroup
+  )(implicit hc: HeaderCarrier): DataEvent = {
+    val auditDetails = Map(
+      "eori"                -> traderProfile.actorId,
+      "affinityGroup"       -> affinityGroup.toString
+    ) ++
```

Comment:
Here you do nor need to use Map concatenation, but just add them as element:

```suggestion
     val auditDetails = Map(
      "eori"                -> traderProfile.actorId,
      "affinityGroup"       -> affinityGroup.toString,
      "previousProfile" -> addUpdatedFieldsOfTraderProfile(traderProfile, updatedTraderProfile),
      "currentProfile" -> addUpdatedFieldsOfTraderProfile(updatedTraderProfile, traderProfile)
    )
```